src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -133,37 +120,46 @@ public final Object runTool(){
     }
 
     public Object instanceMainPostParseArgs() {
-        // Provide one temp directory if the caller didn't
-        if (this.TMP_DIR == null) this.TMP_DIR = new ArrayList<>();
-        if (this.TMP_DIR.isEmpty()) TMP_DIR.add(IOUtil.getDefaultTmpDir());
-
         // Build the default headers
         final ZonedDateTime startDateTime = ZonedDateTime.now();
         this.defaultHeaders.add(new StringHeader(commandLine));
         this.defaultHeaders.add(new StringHeader("Started on: " + Utils.getDateTimeForDisplay(startDateTime)));
 
-        LoggingUtils.setLoggingLevel(VERBOSITY);  // propagate the VERBOSITY level to logging frameworks
+        LoggingUtils.setLoggingLevel(configArgs.getVerbosity());  // propagate the VERBOSITY level to logging frameworks
 
-        for (final File f : TMP_DIR) {
+        for (final Path p : this.configArgs.getResolvedTmpDirectories()) {
             // Intentionally not checking the return values, because it may be that the program does not
             // need a tmp_dir. If this fails, the problem will be discovered downstream.
-            if (!f.exists()) f.mkdirs();
-            f.setReadable(true, false);
-            f.setWritable(true, false);
-            System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
+            if (!Files.exists(p)) {
+                try {
+                    Files.createDirectories(p);
+                } catch (IOException e) {
+                    // intentionally ignoring
+                }
+            }
+            for (final PosixFilePermission perm: Arrays.asList(
+                    PosixFilePermission.OWNER_READ, PosixFilePermission.GROUP_READ, PosixFilePermission.OTHERS_READ,
+                    PosixFilePermission.OWNER_WRITE, PosixFilePermission.GROUP_WRITE, PosixFilePermission.OTHERS_WRITE)) {
+                try {
+                    Files.setPosixFilePermissions(p, Collections.singleton(perm));
```

Comment:
Why loop through the permissions one at a time?  Wouldn't it make sense to pass them all in as a to a single call of Files.setPosixFilePermissions?

Reply:
There are two reasons to implement each permission one by one:
* In the `File` implementation, `File.setReadable` and `File.setWritable`were separated. If one of them failed, the returned value was ignored and the next was applied. With `Files.setPosixFilePermissions` it might fail earlier and thus the same behavior won't be always the case. For example, if the `FileSystem` does not implement readable permissions but just writable ones, perhaps it does not set the writting permissions...
* In the file implementation, the second param (ownerOnly) is false, which determines that the permissions are not only for the user but for everybody. If one of the permissions fail, I am not sure how `Files.setPosixFilePermissions` would behave. The same if the `FileSystem` implementation does throw an error if not owner/group/others is allowed.

I will add at the end of this PR a commit with your solution. Feel free to revert that commit to get this merge as soon as possible to enter in the release.